writers.py

```diff
@@ -0,0 +1,186 @@
+from uuid import uuid4
+import tarfile
+from io import BytesIO
+import gzip
+import json
+import time
+import datetime
+import logging
+import os
+import tempfile
+
+import boto3
+from botocore.exceptions import ClientError
+import pytz
+
+
+LOGGER = logging.getLogger(__name__)
+
+if os.environ.get("LOG_LEVEL"):
+    LOGGER.setLevel(getattr(logging, os.environ.get("LOG_LEVEL")))
+else:
+    LOGGER.setLevel(getattr(logging, "INFO"))
+
+
+class WriteSQL(object):
+    def __init__(self, schema, conn):
+
+        self.schema = schema
+        self.conn = conn
+        self.cursor = None
+
+        self.requests_sql = """INSERT INTO {schema}.requests(data_,
+                        stationid, lineid, all_stations, create_date, pollid, request_date)
+                       VALUES(%(data_)s, %(stationid)s, %(lineid)s, %(all_stations)s, %(create_date)s, %(pollid)s, %(request_date)s)
+                       RETURNING requestid""".format(
+            schema=schema
+        )
+        self.poll_update_sql = """UPDATE {schema}.polls set poll_end = %s
+                        WHERE pollid = %s""".format(
+            schema=schema
+        )
+        self.poll_insert_sql = """INSERT INTO {schema}.polls(poll_start)
+                        VALUES(%s)
+                        RETURNING pollid""".format(
+            schema=schema
+        )
+
+        self.ntas_sql = """INSERT INTO {schema}.ntas_data(\
+            requestid, id, station_char, subwayline, system_message_type, \
+            timint, traindirection, trainid, train_message, train_dest) \
+            VALUES (%(requestid)s, %(id)s, %(station_char)s, %(subwayline)s, %(system_message_type)s, \
+            %(timint)s, %(traindirection)s, %(trainid)s, %(train_message)s, %(train_dest)s);
+          """.format(
+            schema=self.schema
+        )
+
+        self.cursor = self.conn.cursor()
+
+    def add_ntas_record(self, record_row):
+        self.cursor.execute(self.ntas_sql, record_row)
+
+    def add_request_info(self, request_row):
+        self.cursor.execute(self.requests_sql, request_row)
+        request_id = self.cursor.fetchone()[0]
+
+        return request_id
+
+    def add_poll_start(self, time):
+        self.cursor.execute(self.poll_insert_sql, (str(time),))
+        poll_id = self.cursor.fetchone()[0]
+
+        return poll_id
+
+    def add_poll_end(self, poll_id, time):
+        self.cursor.execute(self.poll_update_sql, (str(time), str(poll_id)))
+
+    def commit(self):
+        self.conn.commit()
+        self.cursor.close()
+
+
+class WriteS3(object):
+    def __init__(self, bucket_name):
+        self.ntas_records = []
+        self.requests = []
+        self.polls = {}
+
+        self.bucket_name = bucket_name
+
+        self.s3 = boto3.resource("s3")
+
+    def add_ntas_record(self, record_row):
+        self.ntas_records.append(record_row)
+
+    def add_poll_start(self, time):
+        poll_id = str(uuid4())
```

Comment:
should `uuid1` be used instead so the results are ordered instead of random?

Reply:
Yeah that works as well, I was imagining it as a join key but neither will have any collisions.